File: day05/day5_test.go
package main

import (
	"testing"
)

type Fxn func(s string) string

func TestDay1(t *testing.T) {
	tests := []struct {
		name     string
		fxn      Fxn
		input    string
		expected string
	}{
		{
			name:     "ComputePart1",
			fxn:      ComputePart1,
			input:    input,
			expected: "CMZ",
		},
		{
			name:     "ComputePart2",
			fxn:      ComputePart2,
			input:    input,
			expected: "MCD",
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			res := tt.fxn(tt.input)
			if res != tt.expected {
				t.Errorf("Part1 = %v, expected %v", res, tt.expected)
			} else {
				t.Logf("PASS: %v, %v", tt.name, tt.expected)
			}
		})
	}
}

File: day05/day5.test.js
const aoc = require('./day5')

const input = `\
    [D]    
[N] [C]    
[Z] [M] [P]
 1   2   3 

move 1 from 2 to 1
move 3 from 1 to 3
move 2 from 2 to 1
move 1 from 1 to 2
`
describe('day5', () => {
    test('part1', () => {
        expect(aoc.run({part: 1, input})).toStrictEqual("CMZ")
    })
    test('part2', () => {
        expect(aoc.run({part: 2, input})).toStrictEqual("MCD")
    })
})